server/resourcemanager/resource_manager.go

```diff
@@ -0,0 +1,347 @@
+	}
+
+	bytes, err := encodeValues(resourceList, encoder)
+	if err != nil {
+		writeError(w, encoder, http.StatusInternalServerError, fmt.Errorf("cannot marshal entity: %w", err))
+		return
+	}
+
+	writeResponse(w, 200, string(bytes))
+}
+
+func (m *manager[T]) get(w http.ResponseWriter, r *http.Request) {
+	encoder, err := encoderFromRequest(r)
+	if err != nil {
+		writeResponse(w, http.StatusBadRequest, fmt.Sprintf("cannot process request: %s", err.Error()))
+		return
+	}
+	w.Header().Set("Content-Type", encoder.ResponseContentType())
+
+	vars := mux.Vars(r)
+	id := id.ID(vars["id"])
+
+	item, err := m.handler.Get(r.Context(), id)
+	if err != nil {
+		m.handleResourceHandlerError(w, "getting", err, encoder)
+		return
+	}
+
+	newResource := Resource[T]{
+		Type: m.resourceType,
+		Spec: item,
+	}
+
+	bytes, err := encodeValues(newResource, encoder)
+	if err != nil {
+		writeError(w, encoder, http.StatusInternalServerError, fmt.Errorf("cannot marshal entity: %w", err))
+		return
+	}
+
+	writeResponse(w, 200, string(bytes))
+}
+
+func (m *manager[T]) delete(w http.ResponseWriter, r *http.Request) {
+	encoder, err := encoderFromRequest(r)
+	if err != nil {
+		writeResponse(w, http.StatusBadRequest, fmt.Sprintf("cannot process request: %s", err.Error()))
+		return
+	}
+	w.Header().Set("Content-Type", encoder.ResponseContentType())
+
+	vars := mux.Vars(r)
+	id := id.ID(vars["id"])
+
+	err = m.handler.Delete(r.Context(), id)
+	if err != nil {
+		m.handleResourceHandlerError(w, "deleting", err, encoder)
+		return
+	}
+
+	w.WriteHeader(204)
+}
+
+func (m *manager[T]) handleResourceHandlerError(w http.ResponseWriter, verb string, err error, encoder encoder) {
+	// 404 - not found
+	if errors.Is(err, sql.ErrNoRows) {
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	// 500 - internal server error
+	err = fmt.Errorf("error %s resource %s: %w", verb, m.resourceType, err)
+	writeError(w, encoder, http.StatusInternalServerError, err)
+}
+
+func (m *manager[T]) operationWithBody(w http.ResponseWriter, r *http.Request, statusCode int, operationVerb string, fn func(context.Context, T) (T, error)) {
+	encoder, err := encoderFromRequest(r)
+	if err != nil {
+		writeResponse(w, http.StatusBadRequest, fmt.Sprintf("cannot process request: %s", err.Error()))
+		return
+	}
+	w.Header().Set("Content-Type", encoder.ResponseContentType())
+
+	values, err := readValues(r, encoder)
+	if err != nil {
+		writeError(w, encoder, http.StatusBadRequest, fmt.Errorf("cannot parse body: %w", err))
+		return
+	}
+
+	targetResource := Resource[T]{}
+	err = mapstructure.Decode(values, &targetResource)
+	if err != nil {
+		writeError(w, encoder, http.StatusBadRequest, fmt.Errorf("cannot unmarshal body values: %w", err))
+		return
+	}
+
+	// TODO: if resourceType != values.resourceType return error
+
+	// TODO: check if this needs to be done per operation
+	if !targetResource.Spec.HasID() {
+		targetResource.Spec = m.handler.SetID(targetResource.Spec, m.idgen())
+	}
```

Comment:
question: This enables two different configurations to be saved into the server. In case that happens, which one would be picked and applied by the server?

Reply:
I didn't get to that part yet. These configs are also *not* being consumed by anyone, so it's just to insert/manage the entity. In a later PR, this can be enforced by last created, or always being an update, or other approaches.